### DISCUSSION TITLE
Message expiration not working automatically

### DISCUSSION THREAD

**Discussion by fel486** [rabbitmq/rabbitmq-server#3852] [Other]
I'm running on Docker w/ image "rabbitmq:3.8.14-management"

Publishing a message with expiration to Q1 (which has these headers declared).

```
x-dead-letter-exchange:	my_exchange
x-dead-letter-routing-key: Q2_routing_key
```

When the message on Q1 expires, it won't go to Q2 automatically, I have to go on the Get Message(s) button on Q1 and click it, so the trigger happens.

![image](https://user-images.githubusercontent.com/10427954/144700671-5b309126-3fa5-446d-943b-20af4061de42.png)

I have used RabbitMQ for a time with similar approaches and this has never happened to me. 

Any idea?

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
See [TTL Caveats](https://rabbitmq.com/ttl.html#per-message-ttl-caveats). If head-of-the-line messages are not consumed, expired messages won't be expired. They are expired at (attempted) delivery time.
